From a280bd69c262732250efb5b95e5d0121bb12acf1 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Fri, 22 Nov 2019 08:50:55 +0800
Subject: [PATCH] net: wireless: rockchip: rtl8723cs: fix clang warning

drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/ioctl_cfg80211.c:3876:20: warning: implicit conversion from enumeration type 'enum mlme_auth_type' to different enumeration type 'enum nl80211_auth_type' [-Wenum-conversion]
                sme->auth_type = MLME_AUTHTYPE_SAE;
                               ~ ^~~~~~~~~~~~~~~~~

MLME_AUTHTYPE_SAE == NL80211_AUTHTYPE_SAE, so this fix is safe.

Signed-off-by: Tao Huang <huangtao@rock-chips.com>
Change-Id: Ia3fc1afc6375cc86168c0d9a744e0817df9614c7
---
 .../rockchip_wlan/rtl8723cs/os_dep/linux/ioctl_cfg80211.c       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/ioctl_cfg80211.c b/drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/ioctl_cfg80211.c
index ed18accf652c..3a9eee9b603e 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/ioctl_cfg80211.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/ioctl_cfg80211.c
@@ -3873,7 +3873,7 @@ static int cfg80211_rtw_connect(struct wiphy *wiphy, struct net_device *ndev,
 		sme->privacy, sme->key, sme->key_len, sme->key_idx, sme->auth_type);
 
 	if (rtw_check_connect_sae_compat(sme)) {
-		sme->auth_type = MLME_AUTHTYPE_SAE;
+		sme->auth_type = NL80211_AUTHTYPE_SAE;
 		psecuritypriv->auth_type = MLME_AUTHTYPE_SAE;
 		psecuritypriv->auth_alg = WLAN_AUTH_SAE;
 		RTW_INFO("%s set sme->auth_type for SAE compat\n", __FUNCTION__);
-- 
2.35.3

